OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -270,6 +270,24 @@ void Joint::setChildMobilizedBodyIndex(const SimTK::MobilizedBodyIndex index) co
 }
 
 
+void Joint::extendConnectToModel(Model& model) 
+{
+    Super::extendConnectToModel(model);
+
+    auto& parent = updSocket<PhysicalFrame>("parent_frame").getConnectee();
+    auto& child = updSocket<PhysicalFrame>("child_frame").getConnectee();
+
+    // make sure that parent and child frames are connected to the Model
+    // so we can traverse to base frame
```

Comment:
The comment says "connected to the Model" but we are not invoking `connectToModel()` on the frames. Would it be more accurate to say 

`finalizeConnections()` invokes `extendConnect()` which calls `extendConnectToModel()`; this doesn't seem like the right place to make these calls. Could you end up in an infinite recursion if the parent or child frame owns the joint? Doesn't this mean `finalizeConnections()` will be called multiple times for these frames (once in the normal call sequence across the hierarchy, and once for each joint for which it is a child or parent frame)?